libraries/timeslider/src/components/TimeSlider/TimeSliderModel.ts

```diff
@@ -0,0 +1,264 @@
+            ) {
+                widget.effectiveStops.push(widget.fullTimeExtent.end);
+            }
+        });
+    };
+
+    private _updateWidgetFromWebMapTimeSlider = (
+        widget: EsriTimeSlider,
+        timeSlider: __esri.WebMapTimeSlider,
+        map: WebMap
+    ): void => {
+        let timeExtentOption: string;
+        if (timeSlider.fullTimeExtent) {
+            widget.fullTimeExtent = timeSlider.fullTimeExtent;
+            widget.timeExtent = timeSlider.currentTimeExtent;
+        } else {
+            widget.fullTimeExtent = timeSlider.currentTimeExtent;
+            timeExtentOption = "currentTimeExtent";
+        }
+        // Set stops based on available properties in the time slider
+        // config in this order: StopByDate, StopByCount, StopByInterval
+        if (timeSlider.stops) {
+            widget.stops = {
+                dates: timeSlider.stops,
+                timeExtent: widget.fullTimeExtent,
+            };
+        } else if (timeSlider.numStops) {
+            widget.stops = {
+                count: timeSlider.numStops,
+                timeExtent: widget.fullTimeExtent,
+            };
+        } else if (timeSlider.stopInterval) {
+            widget.stops = {
+                interval: timeSlider.stopInterval,
+                timeExtent: widget.fullTimeExtent,
+            };
+        }
+        // Ensure time slider playback will end on the latest date in the full
+        // time extent.
+        if (
+            widget.effectiveStops?.[widget.effectiveStops.length - 1] <
+            widget.fullTimeExtent.end
+        ) {
+            widget.effectiveStops.push(widget.fullTimeExtent.end);
+        }
+        if (timeExtentOption === "currentTimeExtent") {
+            // Set a default timeExtent if fullTimeExtent was null.
+            widget.timeExtent = new TimeExtent({
+                start: widget.effectiveStops[0],
+                end: widget.effectiveStops[1],
+            });
+        }
+        // Set properties to model from time slider config.
+        if (timeSlider.loop) {
+            this.loop = widget.loop;
+        }
+        if (timeSlider.stopDelay) {
+            this.playRate = widget.playRate;
+        }
+        if (timeSlider.numThumbs === 1) {
+            this.mode = "instant";
+        } else {
+            this.mode = "time-window";
+        }
+        // Need to find the timeVisible boolean inside layer infos - it isn't in
+        // the time slider config.
+        let timeVisible = false;
+        const promises: Promise<void>[] = [];
+        map.layers.forEach((tempLayer) => {
+            const layer = tempLayer as FeatureLayer;
+            promises.push(
+                layer.load().then(() => {
+                    if (layer.timeInfo) {
+                        timeVisible = layer.timeInfo["useTime"];
+                    }
+                })
+            );
+        });
+
+        void Promise.all(promises).then(() => {
```

Comment:
For fire and forget promises like this, I think that a lint override is preferrable (`// eslint-disable-next-line @typescript-eslint/no-floating-promises`) than using the `void` operator, which is bizarre JavaScript esoterica that   few people will know about or use.

Reply:
Fixed.